quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java

```diff
@@ -1106,26 +1106,37 @@ private static void generateDomainAccessorsForShadowSources(IndexView indexView,
                                 ShadowSources.class.getSimpleName()));
             }
             var sources = shadowSources.value().asStringArray();
+            var groupKey = shadowSources.value("groupKey");
+
+            if (groupKey != null && !groupKey.asString().isEmpty()) {
+                generateDomainAccessorsForSourcePath(indexView, rootType, groupKey.asString(),
+                        membersToGeneratedAccessorsForCollection);
+            }
             for (var source : sources) {
-                for (var iterator = RootVariableSource.pathIterator(rootType, source); iterator.hasNext();) {
-                    var member = iterator.next().member();
-                    AnnotationTarget target;
-
-                    if (member instanceof Field field) {
-                        target = indexView.getClassByName(field.getDeclaringClass()).field(field.getName());
-                    } else if (member instanceof Method method) {
-                        target = indexView.getClassByName(method.getDeclaringClass()).method(method.getName());
-                    } else {
-                        throw new IllegalStateException("Member (%s) is not on a field or method."
-                                .formatted(member));
-                    }
-                    // Create a fake annotation for it
-                    membersToGeneratedAccessorsForCollection.add(
-                            AnnotationInstance.builder(DotNames.SHADOW_SOURCES)
-                                    .value(source)
-                                    .buildWithTarget(target));
-                }
+                generateDomainAccessorsForSourcePath(indexView, rootType, source, membersToGeneratedAccessorsForCollection);
+            }
+        }
+    }
+
+    private static void generateDomainAccessorsForSourcePath(IndexView indexView, Class<?> rootType, String source,
+            List<AnnotationInstance> membersToGeneratedAccessorsForCollection) {
+        for (var iterator = RootVariableSource.pathIterator(rootType, source); iterator.hasNext();) {
+            var member = iterator.next().member();
+            AnnotationTarget target;
+
+            if (member instanceof Field field) {
```

Comment:
How would this code decide whether to do the accessor on methods or on the field? The field name is declared in an annotation on another field, so we cannot make the decision based on whether or not the field itself is annotated, as we do elsewhere.

Reply:
The more specific of the two are preferred, using the getter if the types are the same: https://github.com/TimefoldAI/timefold-solver/blob/72d48c258f0cb4130e13c7dfd4cb978844fc7793/core/src/main/java/ai/timefold/solver/core/impl/domain/variable/declarative/RootVariableSource.java#L311-L340